Breeze.UI/src/app/wallet/tumblebit/tumblebit.component.ts

```diff
@@ -117,8 +115,48 @@ export class TumblebitComponent implements OnInit, OnDestroy {
     }
 
     this.stopConnectionRequest();
+
+    if (this.subscriptions) {
+      this.subscriptions.unsubscribe();
+    }
+
+    if (this.router$) {
+      this.router$.unsubscribe();
+      this.router$ = null;
+    }
   };
 
+  private subscribeToRouter(): void {
+    if (this.router$) {
+      this.router$.unsubscribe();
+    }
+    this.router$ = this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
+        if (event.url === this.routerPath) {
+          
+          if (this.subscriptions) {
+            this.subscriptions.unsubscribe();
+          }
+          this.subscriptions = new CompositeDisposable([
```

Comment:
This whole subscription management is a bit tedious. Instead of tracking all subscriptions we could create a helper observable, destroy it in ngOnDestroy and then just append `takeUntil(destroyed$)` prior to every subscribe.

Example:
declaration:
```
destroyed$ = new ReplaySubject<any>();
```
then in ngOnDestroy
```
ngOnDestroy(): void {
        this.destroyed$.next(true);
        this.destroyed$.complete();
    }
```
Then get rid of all subscription tracking and just do this
```
this.apiService.getGeneralInfo(walletInfo)
       .takeUntil(this.destroyed$)
       .subscribe(......
```

Reply:
As per our conversation, this doesn't quite fit with what we are trying to do, and therefore we're going with the original changes.  
The recommendation is interesting and valid, and can play a part in certain scenarios.